src/v/cluster/archival/archival_metadata_stm.cc

```diff
@@ -1050,6 +1050,38 @@ ss::future<> archival_metadata_stm::do_apply(const model::record_batch& b) {
         _manifest->advance_insync_offset(b.last_offset());
         co_return;
     }
+    if (
+      !_raft->log_config().is_read_replica_mode_enabled()
+      && b.last_offset() < _manifest->get_insync_offset()) {
```

Comment:
`<=`?

[1 1] [2 2] are applied, insync offset is 2 now.

if we replay, we need to test for equality here for both to be skipped.

if we test only `<` then first batch will evaluate to true => skipped
second batch will evaluate to false 2 < 2 => replayed **wrong**

Reply:
Test needed pretty please